http_test.go

```diff
@@ -184,3 +184,45 @@ func TestHTTPStreamHandlerHeaderFlushIfNoEvents(t *testing.T) {
 		assert.Fail(t, "Subscribe should returned in 100 milliseconds")
 	}
 }
+
+func TestHTTPStreamHandlerAutoStream(t *testing.T) {
+	t.Parallel()
+
+	sseServer := New()
+	defer sseServer.Close()
+
+	sseServer.AutoStream = true
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/events", sseServer.ServeHTTP)
+	server := httptest.NewServer(mux)
+
+	c := NewClient(server.URL + "/events")
+
+	events := make(chan *Event)
+
+	var cErr error
```

Comment:
You may be better to create a channel for this error, like: 

```go
cErr := make(chan error)

go func() {
	cErr <- c.SubscribeChan("test", events)
}()

require.Nil(t, <-cErr)
```

This is thread safe and avoids having to sleep

You should probably use something like the `wait()` function  though to stop the test from blocking forever however.

Reply:
Done, but it doesn't work. Can you help me here?

Reply:
Is this resolved, @purehyperbole?